### ISSUE TITLE
Integration with Symfony

### ISSUE THREAD

**Issue by drewclauson** [neomerx/limoncello#12]
Hi @neomerx,

I have been trying out a few different json-api PHP libraries and this one seems the most flexible and active!  I've taken a few steps to start using it in simple Symfony controllers, but would like to figure out how to make an integration similar to how you've done the Laravel integration.  I'm available 7am to 6pm in Eastern daylight savings time and could talk tomorrow.

Thanks!
Drew

**Comment by drewclauson** [comment]
I took some time yesterday to play around with this further.  I was able to get a controller almost working using the following setup.  I'm just trying to get modifications the code from limoncello-laravel ported over a little.

```
class SandboxController extends Controller
{

    use JsonApiTrait;
    use AppServiceProviderTrait;
    /**
     * @Method("GET")
     * @Route("/sandbox")
     * @return \Symfony\Component\HttpFoundation\Response
     */
    public function indexAction()
    {

        $integration = $this->container->get('rest.symfony_integration');
        $this->registerResponses($integration);
        $this->registerCodecMatcher($integration);
        $this->registerExceptionThrower($integration);
        $this->container->set(FactoryInterface::class, new Factory());
        $this->container->set(CodecMatcherInterface::class, new CodecMatcher());
        $this->container->set(ExceptionThrowerInterface::class, new ExceptionThrower());


        $this->initJsonApiSupport($integration);

        $this->checkParametersEmpty();

        $em = $this->getDoctrine()->getManager();
        $data = $em->getRepository('CustomerBundle:Customer')->findAllTest();

        return $this->getContentResponse($data, Response::HTTP_OK, null, null);

    }
}

```

**Comment by drewclauson** [comment]
That controller is working up until line 40 of the Laravel(Symfony)Integration file where it tries to:

```
$config = config(C::NAME, []);
```

For whatever reason, I'm not certain what the equivalent would be in Symfony.

**Comment by drewclauson** [comment]
What time works best for you to discuss?  Were you thinking chat or some sort of call?

**Comment by neomerx** [comment]
@drewclauson Login to gitter :wink:

**Comment by drewclauson** [comment]
Just to note anyone finding this, I'm working on an integration bundle for Symfony and will post back here once I put it out online.

**Comment by drewclauson** [comment]
I've successfully created an integration package at https://github.com/drewclauson/elytus-limoncello-bundle.  Documentation and test are forthcoming.

**Comment by drewclauson** [comment]
And feedback is very welcome as it's my first bundle that I've made publicly available.

**Comment by neomerx** [comment]
@drewclauson Congrats! As I don't know symfony I've got not much to say :smile:
I think it would be nice to have some 'how-to install/use' description, travis-ci configured and then the link should be posted to http://jsonapi.org/implementations/#server-libraries-php after limoncello-collins (dont add it to the end :wink: )

**Comment by drewclauson** [comment]
@neomerx Thanks for the suggestions!  All of those things I've got in the plans, but it may take me a few weeks to get there.  I'm in the process of learning about how to use JsonApiTrait in limoncello and then I'll get back to updating the docs and make a few more tweaks.

**Comment by neomerx** [comment]
Most important is a usage sample. Tests could be added later with no probs IMO. The trait adds to controller a few method (e.g.)
- checkParametersEmpty
- getParameters
- getCodeResponse
- getMetaResponse
- getContentResponse
- getCreatedResponse

Just try one of them. For example,
```php
$symfonyResponse = $this->getContentResponse($arrayOfDoctrineModels);
```

**Comment by drewclauson** [comment]
Thanks, I'll be working on the usage sample as I get more familiar with how the trait controller methods work.  

Question: on the trait, there are several settings like `protected $allowedPagingParameters = [];`  In order to get these to work in Laravel, do you override them on the controller?  I noticed that you have to call them before you call 'initJsonApiSupport' or they won't be recognized.  Do you override them, set them in the constructor, or something else in your framework?  If I override them or put them in the constructor, then those settings have to be used for every controller in the class, so I'm probably going to create an annotation driver to be able to configure them via annotations per controller method.

For example, I have FooController which has the methods `indexAction()` (collection of Foos) and `showAction()` (single Foo).  If I override the `allowedPagingParameters` to allow 'limit' and 'offset' in the FooController, then the setting applies for both index and show.  Thus, paging parameter is enabled on `showAction`.  While it won't cause a problem, isn't ideal IMHO, as `showAction` probably shouldn't have the parameter enabled.  So, all that to say, how do you handle this in Laravel and I'll translate into Symfony to the best of my ability? :-)

**Comment by neomerx** [comment]
Have you seen [this](https://github.com/neomerx/limoncello-collins#adding-a-new-service) and esp. [this](https://github.com/neomerx/json-api/wiki)?

**Comment by drewclauson** [comment]
Yes, but I'm talking specifically about limoncello and [this parameter as an example] (https://github.com/neomerx/limoncello/blob/master/src/Http/JsonApiTrait.php#L89).

In order to allow the paging parameters 'limit' and 'offset' to be used without throwing an error when `$this->checkParameters()` runs, I have to configure:

`$allowedPagingParameters = ['limit','offset'];`

This needs to be done before calling `$this->initJsonApiSupport($integration)`, correct?

**Comment by neomerx** [comment]
It's designed to be used this way
- include the trait to base Controller class (e.g. `MyAppBaseController`)
- set parameters in particular controller. For example
```php
class SandboxController extends MyAppBaseController
{
    $allowedPagingParameters = ['limit','offset'];

    ...
```

**Comment by drewclauson** [comment]
Haha, that was the one controller I didn't look at... Thanks!

**Comment by nvshivakumar** [comment]
Hi need help to integrate neomerx with symfony2

**Comment by drewclauson** [comment]
@nvshivakumar Hi, I have created an integration bundle at https://github.com/drewclauson/elytus-limoncello-bundle, however I have not been able to put together a good example project due to deadline constraints at work right now.  You're welcome to take a look and make any suggestions, and I'll try to have some tutorial information posted in the next week or so.  I'm sorry I can't be of immediate assistance.

**Comment by nvshivakumar** [comment]
I have n't see any update
Btw I need how to filter queryresponses which is available at Wooho
